### PR TITLE
Create datasource azuredevops_serviceendpoint_github, add common func…

### PR DESCRIPTION
**Pull request by a30004053**
…tions to create datasource schemas, fix documentation

## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

This PR adds a new data source `azuredevops_serviceendpoint_github` which returns details of a GitHub service endpoint defined by an `id` or a unique `service_endpoint_name`

Issue Number: #608

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Any relevant logs, error output, etc?

https://pst.klgrth.io/paste/4t6et

## Other information

Additional functions are added to `azuredevops/internal/service/serviceendpoint/commons.go`  to support common dataSource schemas and serviceendpoint lookup functions

Modifications were made to `azuredevops_serviceendpoint_azurerm` datasource as a result of the aforementioned additions to `commons.go`

### CODE REVIEW

**Comment by a30004053** [comment]
`TestAccServiceEndpointGitHub_with_serviceEndpointName_DataSource`
```bash
TF_ACC=1 go test -tags "all" $(go list ./azuredevops/internal/acceptancetests |grep -v 'vendor') -v -run=TestAccServiceEndpointGitHub_with_serviceEndpointName_DataSource -timeout 120m
=== RUN   TestAccServiceEndpointGitHub_with_serviceEndpointName_DataSource
=== PAUSE TestAccServiceEndpointGitHub_with_serviceEndpointName_DataSource
=== CONT  TestAccServiceEndpointGitHub_with_serviceEndpointName_DataSource
--- PASS: TestAccServiceEndpointGitHub_with_serviceEndpointName_DataSource (61.09s)
PASS
ok      github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/acceptancetests        61.205s
```

`TestAccServiceEndpointGitHub_with_serviceEndpointID_DataSource`
```bash
TF_ACC=1 go test -tags "all" $(go list ./azuredevops/internal/acceptancetests |grep -v 'vendor') -v -run=TestAccServiceEndpointGitHub_with_serviceEndpointID_DataSource -timeout 120m
=== RUN   TestAccServiceEndpointGitHub_with_serviceEndpointID_DataSource
=== PAUSE TestAccServiceEndpointGitHub_with_serviceEndpointID_DataSource
=== CONT  TestAccServiceEndpointGitHub_with_serviceEndpointID_DataSource
--- PASS: TestAccServiceEndpointGitHub_with_serviceEndpointID_DataSource (51.51s)
PASS
ok      github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/acceptancetests        51.625s
```

**Comment by a30004053** [comment]
Hello @xuzhang3, 

Hope you are well. 
When you get a chance can you please review this PR?

Cheers!

File: website/docs/d/serviceendpoint_github.html.markdown [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+layout: "azuredevops"
+page_title: "AzureDevops: Data Source: azuredevops_serviceendpoint_github"
+description: |-
+  Gets information about an existing AzureRM Service Endpoint.
```
Comment (xuzhang3) [inline website/docs/d/serviceendpoint_github.html.markdown]:
```suggestion
    Gets information about an existing GitHub Service Endpoint.
```

File: azuredevops/internal/service/serviceendpoint/commons.go [hunk 2]
```diff
@@ -381,3 +381,130 @@ func getServiceEndpoint(client *client.AggregatedClient, serviceEndpointID *uuid
...
+
+			"service_endpoint_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ExactlyOneOf: []string{"service_endpoint_name", "id"},
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"id": {
```
Comment (xuzhang3) [inline azuredevops/internal/service/serviceendpoint/commons.go]: can we rename the `id` to  to `serviceendpoint_id`,   Terraform resource have a default `id` property  same name may result in unexpected behavior.

**Review by xuzhang3 — CHANGES_REQUESTED** [review]
@a30004053   thanks for your contribute. A few change  requested, others look good to me.

**Comment by a30004053** [comment]
Hello @xuzhang3, 

If it all looks okay from your end, can this PR be merged?

Thanks in advance!

**Comment by xuzhang3** [comment]
@a30004053 LGTM

**Review by xuzhang3 — APPROVED** [review]

**Pull request merged**